industry_benchmarks/utils/tape_networks.py

```diff
@@ -0,0 +1,459 @@
+                    units[0]["inputs"]["stateA"]
+                )
+                stateB = ChemicalSystem.from_dict(
+                    units[0]["inputs"]["stateB"]
+                )
+                ligmap = LigandAtomMapping.from_dict(
+                    units[0]["inputs"]["ligandmapping"]
+                )
+            else:  # inputs were cleaned out! reengineer - hacky alchem net.
+                if input_ligand_network is not None:
+
+                    transfromation_name = (
+                        units[0]["name"].split(" repeat")[0].strip()
+                    )
+                    componentA_name, componentB_name = (
+                        transfromation_name.split(" to ")
+                    )
+
+                    ligmap = None
+                    for e in input_ligand_network.edges:
+                        if all(
+                            cn in (e.componentA.name, e.componentB.name)
+                            for cn in [componentA_name, componentB_name]
+                        ):
+                            ligmap = e
+                            break
+                    if ligmap is None:
+
+                        raise ValueError(
+                            "could not finde edge: ("
+                            + e.componentA.name
+                            + ", "
+                            + e.componentB.name
+                            + ")"
+                        )
+                    else:
+                        if "solvent" in str(units[0]["outputs"]["nc"]):
+                            stateA = ChemicalSystem(
+                                components={
+                                    "ligand": ligmap.componentA,
+                                    "solvent": SolventComponent(),
+                                }
+                            )
+                            stateB = ChemicalSystem(
+                                components={
+                                    "ligand": ligmap.componentB,
+                                    "solvent": SolventComponent(),
+                                }
+                            )
+                        else:
+                            stateA = ChemicalSystem(
+                                components={"ligand": ligmap.componentA}
+                            )
+                            stateB = ChemicalSystem(
+                                components={"ligand": ligmap.componentB}
+                            )
+                else:
+                    raise ValueError(
+                        "Can not read alchemical input, as components are missing, please provied input ligand_network."
+                    )
+
+            # Count if both transformations present.
+            if name not in edge_count:
+              edge_count[name] = {"protein": 0, "solvent":0}
+                
+            if (
+                "protein" in stateA.components
+                or "solvent" not in stateA.components
```

Comment:
This line I don't quite understand yet. Is that for the vacuum part?

Reply:
na this is a hack. 
if the components of the result json were removed due to clean up, I infer the edge type from the edge name.
if it is a solvent edge I add a solvent component.
if it is a complex ex I don't do anything, so only a ligand will be present. 

This comes from when I started this, the input alchemical network was not present and therefore the protein component was out of reach. Potentially we could change that, not sure about the gain.